### ISSUE TITLE
Error when using HDF5 data

### ISSUE THREAD

**Issue by Nateliu712** [LisaAnne/lisa-caffe-public#5]
Hi,
I have been trying to train a model using HDF5 data, but I get following error
I1026 22:00:46.879263 23442 caffe.cpp:113] Use GPU with device ID 0
I1026 22:00:47.014865 23442 caffe.cpp:121] Starting Optimization
I1026 22:00:47.014964 23442 solver.cpp:32] Initializing solver from parameters: 
test_iter: 100
test_interval: 100
base_lr: 0.001
display: 20
max_iter: 30000
lr_policy: "step"
gamma: 0.1
momentum: 0.9
weight_decay: 0.005
stepsize: 10000
snapshot: 5000
snapshot_prefix: "simple_lstm_SRL"
solver_mode: GPU
device_id: 0
random_seed: 1701
net: "./examples/SRL/train_test_lstm_SRL.prototxt"
test_state {
  stage: "test-on-test"
}
average_loss: 1000
clip_gradients: 5
I1026 22:00:47.015046 23442 solver.cpp:70] Creating training net from net file: ./examples/SRL/train_test_lstm_SRL.prototxt
I1026 22:00:47.015287 23442 net.cpp:258] The NetState phase (0) differed from the phase (1) specified by a rule in layer data
I1026 22:00:47.015308 23442 net.cpp:258] The NetState phase (0) differed from the phase (1) specified by a rule in layer data
I1026 22:00:47.015319 23442 net.cpp:291] The NetState did not contain stage 'lstm-drop' specified by a rule in layer lstm-drop
I1026 22:00:47.015331 23442 net.cpp:258] The NetState phase (0) differed from the phase (1) specified by a rule in layer accuracy
I1026 22:00:47.015389 23442 net.cpp:42] Initializing net from parameters: 
name: "lstm_SRL_model"
state {
  phase: TRAIN
}
layer {
  name: "data"
  type: "HDF5Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  hdf5_data_param {
    source: "./examples/SRL/hdf5/36p/files_name36p.txt"
    batch_size: 20
  }
}
layer {
  name: "lstm1"
  type: "LSTM"
  bottom: "data"
  bottom: "label"
  top: "lstm1"
  recurrent_param {
    num_output: 256
    weight_filler {
      type: "uniform"
      min: -0.08
      max: 0.08
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "predict"
  type: "InnerProduct"
  bottom: "lstm1"
  top: "predict"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 100
    weight_filler {
      type: "uniform"
      min: -0.08
      max: 0.08
    }
    bias_filler {
      type: "constant"
      value: 0
    }
    axis: 2
  }
}
layer {
  name: "cross_entropy_loss"
  type: "SoftmaxWithLoss"
  bottom: "predict"
  bottom: "label"
  top: "cross_entropy_loss"
  loss_weight: 20
  loss_param {
    ignore_label: -1
  }
  softmax_param {
    axis: 2
  }
}
I1026 22:00:47.015642 23442 layer_factory.hpp:74] Creating layer data
I1026 22:00:47.015671 23442 net.cpp:84] Creating Layer data
I1026 22:00:47.015684 23442 net.cpp:339] data -> data
I1026 22:00:47.015712 23442 net.cpp:339] data -> label
I1026 22:00:47.015725 23442 net.cpp:113] Setting up data
I1026 22:00:47.015734 23442 hdf5_data_layer.cpp:80] Loading list of HDF5 filenames from: ./examples/SRL/hdf5/36p/files_name36p.txt
I1026 22:00:47.019219 23442 hdf5_data_layer.cpp:94] Number of HDF5 files: 14400
HDF5-DIAG: Error detected in HDF5 (1.8.11) thread 140490168145792:
  #000: ../../../src/H5F.c line 1586 in H5Fopen(): unable to open file
    major: File accessibilty
    minor: Unable to open file
  #001: ../../../src/H5F.c line 1275 in H5F_open(): unable to open file: time = Mon Oct 26 22:00:47 2015
, name = './examples/SRL/hdf5/14p/P15_01_01_0._body.h5', tent_flags = 0
    major: File accessibilty
    minor: Unable to open file
  #002: ../../../src/H5FD.c line 987 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #003: ../../../src/H5FDsec2.c line 343 in H5FD_sec2_open(): unable to open file: name = './examples/SRL/hdf5/14p/P15_01_01_0._body.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0
    major: File accessibilty
    minor: Unable to open file
F1026 22:00:47.019774 23442 hdf5_data_layer.cpp:32] Failed opening HDF5 file: ./examples/SRL/hdf5/14p/P15_01_01_0._body.h5

Is there someong can help me
